parcels/field.py

```diff
@@ -1339,11 +1352,13 @@ def __init__(self, name, U, V, W=None):
         self.V = V
         self.W = W
         self.vector_type = '3D' if W else '2D'
+        self.gridindexingtype = U.gridindexingtype
         if self.U.interp_method == 'cgrid_velocity':
             assert self.V.interp_method == 'cgrid_velocity', (
                 'Interpolation methods of U and V are not the same.')
-            assert self.U.grid is self.V.grid, (
-                'Grids of U and V are not the same.')
+            if U.gridindexingtype == 'nemo':
+                assert self.U.grid is self.V.grid, (
+                    'Grids of U and V are not the same.')
```

Comment:
Since we chose not to support uneven grids for mitgcm after all, this change is not needed. If-statement can be removed